lib/core/src/sync/mod.rs

```diff
@@ -208,6 +227,124 @@ impl SyncService {
+                    sync_data,
+                    sync_state.record_revision,
+                    self.signer.clone()
+                ),
+                "Could not create new record: {}"
+            );
+
+            // Step 5: Push the record
+            let req = unwrap_or_continue!(
+                SetRecordRequest::new(record, utils::now(), self.signer.clone()),
+                "Could not create SetRecordRequest: {}"
+            );
+            let reply =
+                unwrap_or_continue!(self.client.push(req).await, "Could not push record: {}");
+
+            // Step 6: Check for conflict. If present, skip and retry on the next call
+            if reply.status() == SetRecordStatus::Conflict {
+                continue;
+            }
+
+            // Step 7: Update state revision
+            unwrap_or_continue!(
+                self.persister.set_sync_state(SyncState {
+                    data_id: sync_state.data_id,
+                    record_id: sync_state.record_id,
+                    record_revision: reply.new_revision,
+                    is_local: sync_state.is_local,
+                }),
+                "Could not set final sync state after push: {}"
+            );
+        }
+
+        Ok(())
+    }
+
+    pub(crate) async fn enqueue_record(
+        &self,
+        data_id: &str,
+        record_type: RecordType,
+        new_updated_fields: Option<Vec<String>>,
+    ) -> Result<()> {
+        let maybe_previous_sync_state = self.persister.get_sync_state_by_data_id(data_id)?;
+
+        let record_revision = maybe_previous_sync_state
+            .as_ref()
+            .map(|sync_state| sync_state.record_revision)
+            .unwrap_or(0);
+        let is_local = maybe_previous_sync_state
+            .as_ref()
+            .map(|sync_state| sync_state.is_local)
+            .unwrap_or(true);
+        let record_id = maybe_previous_sync_state
+            .map(|sync_state| sync_state.record_id)
+            .unwrap_or(uuid::Uuid::new_v4().to_string());
+
+        // Try to merge previous updated fields with new ones, if either is present
+        let maybe_previous_updated_fields = self
+            .persister
+            .get_sync_outgoing_details_by_id(&record_id)?
+            .map(|details| details.updated_fields)
+            .flatten();
+
+        let updated_fields = match (maybe_previous_updated_fields, new_updated_fields) {
+            (None, None) => None,
+            (Some(previous_fields), None) => Some(previous_fields),
+            (None, Some(new_fields)) => Some(new_fields),
+            (Some(previous_fields), Some(new_fields)) => {
+                Some([previous_fields, new_fields].concat())
+            }
+        };
+
+        let outgoing_details = SyncOutgoingDetails {
+            record_id: record_id.clone(),
+            record_type,
+            updated_fields,
+            commit_time: utils::now(),
+        };
+        let new_state = SyncState {
```

Comment:
As far as I understand the enqueue method should update the update_fields in the sync_outgoing_details so why do we need to update the sync_state ?